backend/data_preparation/crawler/fire_crawler.py

```diff
@@ -0,0 +1,318 @@
+        response = requests.get(url).content.decode("utf-8")
+        logger.info(f"Finished fetching {page_name} page in function {function_name}")
+        return response
+
+    @staticmethod
+    @retry(stop_max_attempt_number=3)
+    def _download_single_file(url: str, file: str, out_path: str):
+        """
+        Downloads a single file from the website to the out_path.
+        :param url: url of the fire event.
+               e.g. https://rmgsc.cr.usgs.gov/outgoing/GeoMAC/2017_fire_data/California/Eclipse/
+        :param file: file name of the file to download.
+                e.g. 'ca_eclipse_20170823_0000_dd83.cpg'
+        :param out_path: path where the downloaded file stored.
+                e.g. '/myResearch/Wildfires/data/fire-data'
+        """
+        logger.info(f"Downloading {url}/{file} to {out_path}.")
+        wget.download(url=url + "/" + file, out=out_path)
+        logger.info(f"Finished Downloading {url}/{file} to {out_path}.")
+
+    @staticmethod
+    @retry(retry_on_exception=requests.exceptions.RequestException, stop_max_attempt_number=3)
+    def _extract_fire_events_in_state(year: int, state: str) -> List[FireEvent]:
+        """
+        Takes a year, and a state, fetches all fire events in that year and the state and output it as a list of
+        FireEvent objects.
+        :param year: int, represent the year to crawl.
+                e.g. 2017
+        :param state: str, the state to crawl.
+                e.g. 'California'
+        :return: list of FireEvent objects of all fire events happened in that year this state
+                e.g. [FireEvent(-1, 2015, 'California', 'FireA'), FireEvent(-1, 2015, 'California', 'FireB')]
+        """
+        logger.info(f"Fetching fire events in {state} in year {year}.")
+        current_year = datetime.datetime.now().date().year
+        url_year = "current_year" if current_year == year else year
+        fire_events_page_in_state = FireCrawler._get_url(f"{FireCrawler.BASE_DIR}{url_year}_fire_data/{state}",
+                                                         f"{state} State of year {year}",
+                                                         "_extract_fire_events_in_state()")
+        fire_names_in_this_state = FireCrawler.RE_EXTRACT_FIRE_EVENTS_IN_STATE.findall(fire_events_page_in_state)
+        logger.info(f"Finished fetching fire events in {state} in year {year}.")
+        return list(map(lambda single_fire_name: FireEvent(year, state, single_fire_name), fire_names_in_this_state))
+
+    @staticmethod
+    def _filter_out_invalid_fire_events(fires: List[FireEvent]):
+        """
+        Find those invalid FireEvent objects and remove them from the input list
+        :param fires: list of FireEvent objects.
+                e.g. [FireEvent(-1, 2015, 'California', 'FireA'), FireEvent(-1, 2015, 'California', 'FireB')]
+        """
+        for fire_event in fires[:]:
+            if fire_event.is_valid():
```

Comment:
sorry I don't understand the logic. the method description seems to tend to filter out the invalid ones, but here the if statement is looking for `is_valid()` ones?